src/main/java/com/fasterxml/jackson/core/json/UTF8JsonGenerator.java

```diff
@@ -2249,8 +2249,8 @@ private byte[] getHexBytes() {
     }
 
     // @since 2.18
-    private boolean _isSurrogateChar(int ch) {
-        return (ch & 0xD800) == 0xD800;
+    private static boolean _isStartOfSurrogatePair(final int ch) {
+        return (ch & 0xF800) == 0xD800;
```

Comment:
I made this change based on comments in #1360 but would like to have it reviewed

Reply:
Updated my comment: I think mask actually needs to be `0xFC00` for First surrogate (i.e. not match Second part of surrogate pair). Will change.